tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,931 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3be0fafd",
+   "metadata": {},
+   "source": [
+    "# TensorDictModule"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94bd315a",
+   "metadata": {},
+   "source": [
+    "We recommand reading the TensorDict tutorial before going through this one."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0652352c",
+   "metadata": {},
+   "source": [
+    "For a convenient usage of the `TensorDict` class with `nn.Module`, TorchRL provides an interface between the two named `TensorDictModule`. <br/>\n",
+    "The `TensorDictModule` class is an `nn.Module` that takes a `TensorDict` as input when called. <br/>\n",
+    "It is up to the user to define the keys to be read as input and output."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5b0241ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "\n",
+    "from torchrl.data import TensorDict\n",
+    "from torchrl.modules import TensorDictModule, TensorDictSequence"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "129a6de9-cf97-4565-a229-c05ad18df882",
+   "metadata": {},
+   "source": [
+    "## `TensorDictModule` by examples\n",
+    "Let's learn about `TensorDictModule by exploring some examples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d1c188a",
+   "metadata": {},
+   "source": [
+    "### Example 1: Simple usage"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1d21a711",
+   "metadata": {},
+   "source": [
+    "Let's imagine we have 2 entries `TensorDict`, a and b and we only want to pass a to our network."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "6f33781f",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "TensorDict(\n",
+       "    fields={\n",
+       "        a: Tensor(torch.Size([5, 3]), dtype=torch.float32),\n",
+       "        a_out: Tensor(torch.Size([5, 10]), dtype=torch.float32),\n",
+       "        b: Tensor(torch.Size([5, 4, 3]), dtype=torch.float32)},\n",
+       "    batch_size=torch.Size([5]),\n",
+       "    device=cpu,\n",
+       "    is_shared=False)"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tensordict = TensorDict(\n",
+    "    {\"a\": torch.randn(5, 3), \"b\": torch.zeros(5, 4, 3)}, batch_size=[5]\n",
+    ")\n",
+    "linear = TensorDictModule(nn.Linear(3, 10), in_keys=[\"a\"], out_keys=[\"a_out\"])\n",
+    "linear(tensordict)\n",
+    "assert (tensordict[\"b\"] == torch.zeros(5, 4, 3)).all()\n",
```

Comment:
is this necessary? You could also say 
```python
assert (tensordict["b"] == 0).all()
```
Also I would suggest to use `get(key)` rathre than indexing with tensordict whenever possible :)